File: gitdot-web/packages/api/src/endpoint/review/update-diff.ts
import { z } from "zod";
import { ReviewResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const UpdateDiffRequest = z.object({
  title: z.string().optional(),
  description: z.string().optional(),
});
export type UpdateDiffRequest = z.infer<typeof UpdateDiffRequest>;

export const UpdateDiffResponse = ReviewResource;
export type UpdateDiffResponse = z.infer<typeof UpdateDiffResponse>;

export const UpdateDiff = {
  path: "/repository/{owner}/{repo}/review/{number}/diff/{position}",
  method: "PATCH",
  request: UpdateDiffRequest,
  response: UpdateDiffResponse,
} as const satisfies Endpoint;
export type UpdateDiff = typeof UpdateDiff;

File: gitdot-api/src/endpoint/review/update_diff.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::review::ReviewResource};

pub struct UpdateDiff;

impl Endpoint for UpdateDiff {
    const PATH: &'static str = "/repository/{owner}/{repo}/review/{number}/diff/{position}";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateDiffRequest;
    type Response = UpdateDiffResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct UpdateDiffRequest {
    pub title: Option<String>,
    pub description: Option<String>,
}

pub type UpdateDiffResponse = ReviewResource;
